sdk/golang/ccnp/quote/quote.go

```diff
@@ -0,0 +1,191 @@
+/*
+* Copyright (c) 2023, Intel Corporation. All rights reserved.<BR>
+* SPDX-License-Identifier: Apache-2.0
+ */
+
+package quote
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"encoding/binary"
+	"log"
+	"strings"
+	"time"
+
+	pb "github.com/intel/confidential-cloud-native-primitives/sdk/golang/ccnp/quote/proto"
+	pkgerrors "github.com/pkg/errors"
+	"google.golang.org/grpc"
+)
+
+const (
+	UDS_PATH = "unix:/run/ccnp/uds/quote-server.sock"
+	TYPE_TDX = "TDX"
+	TYPE_TPM = "TPM"
+)
+
+type TPMQuote struct{}
+
+type TDXQuote struct {
+	Quote           []uint8    // full TD quote
+	Version         uint16     // TD quote version
+	Tdreport        [584]uint8 // full TD report
+	Tee_type        uint32     // Type of TEE for which the Quote has been generated
+	Tee_tcb_svn     [16]uint8  // Array of TEE TCB SVNs
+	Mrseam          [48]uint8  // Measurement of the SEAM module (SHA384 hash)
+	Mrsignerseam    [48]uint8  // Measurement of a 3rd party SEAM module’s signer (SHA384 hash)
+	Seam_attributes [8]uint8   // ATTRIBUTES of SEAM
+	Td_attributes   [8]uint8   // ATTRIBUTES of TD
+	Xfam            [8]uint8   // XFAM of TD
+	Mrtd            [48]uint8  // Measurement of the initial contents of the TD (SHA384 hash)
+	Mrconfigid      [48]uint8  // Software defined ID for non-owner-defined configuration of the TD
+	Mrowner         [48]uint8  // Software defined ID for the guest TD’s owner
+	Mrownerconfig   [48]uint8  // Software defined ID for owner-defined configuration of the TD
+	Rtmrs           [192]uint8 // Array of 4 runtime extendable measurement registers (SHA384 hash)
+	Report_data     [64]uint8  // Additional Report Data
+	Signature       [64]uint8  // ECDSA signature, r component followed by s component, 2 x 32 bytes
+	Attestation_key [64]uint8  // Public part of ECDSA Attestation Key generated by Quoting Enclave
+	Cert_data       []uint8    // Data required to certify Attestation Key used to sign the Quote
+}
+
+// definition of the quote header and TDReport at:
+// https://github.com/intel/SGXDataCenterAttestationPrimitives/blob/6882afad8644c27db162b40994402c8ad2a7fb32/QuoteGeneration/quote_wrapper/common/inc/sgx_quote_4.h#L141
+type SGX_Quote_Header struct {
+	Version      uint16    // The version this quote structure.
+	Att_key_type uint16    // sgx_attestation_algorithm_id_t.  Describes the type of signature in the signature_data[] field.
```

Comment:
Not sure if follow golang naming BKM here, shall "AttKeyType" be better?  Similar comments as others.

Reply:
Just to be sync with header naming

Reply:
> Not sure if follow golang naming BKM here, shall "AttKeyType" be better? Similar comments as others.

Agree on this. In the spec, all attributes are in capital. Can we format these attributes using mixCaps or MixCaps following the naming convention here? https://www.mohitkhare.com/blog/go-naming-conventions/

Reply:
updated